src/HttpOverStream.Client/DialMessageHandler.cs

```diff
@@ -125,8 +122,9 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
                         throw new HttpRequestException("Error parsing header", ex);
                     }
                 }
-                Debug.WriteLine("Client: Finished reading response lines");
+                Debug.WriteLine("Client: Finished reading response header lines");
                 responseContent.SetContent(new BodyStream(stream, response.Content.Headers.ContentLength, closeOnReachEnd: true), response.Content.Headers.ContentLength);
+                writeContentCTS.Cancel();

```

Comment:
SetContent does not read the content,... I am not sure if we can rely on the closeOnReachEnd though...
The more I think about it, the less I think the writerCancelToken is usefull: we have no way here to detect that the user is done reading the response (it happens when she consumes the response HttpContent), so I guess we can get rid of this one, (just pass the timeout-based cancel token as is) and everyone should be happy.

Reply:
Ok, I see what you mean. I guess there is nothing to ensure that a client will cancel the cancellation token though right? In which case, if the client sends a body which doesnt get read, this will leak never ending client streams which are just waiting for their bodies to be read.